lib/iris/tests/__init__.py

```diff
@@ -91,6 +92,13 @@
     GRIB_AVAILABLE = True
 
 
+#: CDL history attribute regular expression pattern for ISO8601 timestamp.
+iso8601 = '(?P<lhs>\s*:history = ")' \
+    '(?P<timestamp>\d{4}\-?\d{2}\-?\d{2}' \
+    '(T\d{2}\:?\d{2}\:?\d{2}(\.\d{3})?(([\+\-]\d{2}\:?\d{2})|Z)?)?: )' \
+    '(?P<rhs>.*)'
```

Comment:
Why not use `datetime.strptime` to do this for you? That way if whatever you pass to `strptime` can't be cast to the format string you also pass it then you get a nice error from the `datetime` module?